mosaic-runtime/src/commonMain/kotlin/com/jakewharton/mosaic/canvas.kt

```diff
@@ -19,25 +19,58 @@ internal interface TextCanvas {
 	operator fun get(row: Int, column: Int): TextPixel
 }
 
-private val blankPixel = TextPixel(' ')
-
 internal class TextSurface(
-	override val width: Int,
-	override val height: Int,
+	initialWidth: Int,
+	initialHeight: Int,
 ) : TextCanvas {
+	private var realWidth: Int = initialWidth
+	private var realHeight: Int = initialHeight
+
+	override val width: Int get() = realWidth
+	override val height: Int get() = realHeight
+
 	override var translationX = 0
 	override var translationY = 0
 
-	private val rows = Array(height) { Array(width) { TextPixel(' ') } }
+	private val rows = MutableList(height) { createBlankRow(width) }
+
+	override operator fun get(row: Int, column: Int): TextPixel {
+		val x = translationX + column
+		val y = translationY + row
+		if (x < 0 || y < 0) {
+			return reusableDirtyPixel
+		}
+		val widthDiff = x - width + 1
+		if (widthDiff > 0) {
+			if (widthDiff == 1) {
+				rows.forEach { it.add(newBlankPixel) }
+			} else {
+				rows.forEach { it.addAll(createBlankRow(widthDiff)) }
```

Comment:
I think this is not an ideal solution, because several lists are created, but on the other hand it should not be a hot execution path.